### PR TITLE
build: add linter to make code style closer to Node.js'

### LINKED ISSUE #3: What is our linter preference
See: https://github.com/pkgjs/parseargs/pull/1#discussion_r615043617

### PR DESCRIPTION
**Pull request by bcoe**
adds a linter that enforces a coding style closer to Node core's.

_Note: this will make it easy to drop code directly into the codebase when the time comes._

Fixes #3

### CODE REVIEW

File: index.js
Comment (bcoe) [inline index.js:58]:
```suggestion
        // isn't used ie. both --foo=b and --foo b should work
```

**Review by iansu — APPROVED** [review]
Nice! Do we want to publish the eslint plugin and/or move it to the pkgjs org before merging this?

**Comment by bcoe** [comment]
@iansu perhaps that's worth doing, @mhdawson are you able to transfer @iansu's eslint preset into pkgjs, and give his npm user access to the organization on npm?

Alternatively, if you gave `oss-bot` write access to the org, I could setup automated publication, like I have for yargs, see:

https://www.npmjs.com/package/yargs

**Comment by bcoe** [comment]
_Note: we should probably opt for `@pkgjs/eslint-plugin-node-core`, rather than `eslint-plugin-node-core`, as `eslint-plugin-node-core` was published a few years ago by Joyee.

**Review by Eomm — APPROVED** [review]

**Review by joesepi — APPROVED** [review]
LGTM. Thanks.

File: index.js
Comment (bcoe) [reply index.js:14]: I think so, this edit was made automatically by linter.

**Comment by bcoe** [comment]
@iansu we can follow up getting rid of the git dependency.

**Pull request merged**
